bluesky/callbacks/waterfall.py

```diff
@@ -0,0 +1,168 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.widgets import Slider
+
+
+class Waterfall:
+    """class holds data and generate watefall plot
+
+    Parameters
+    ----------
+    fig : matplotlib.Figure
+        fig this waterfall plot will be drawn on
+    canvas : matplotlib.Canvas
+        canvas this waterfall plot will be drawn on
+    key_list : list, optional
+        list of key names. default to None
+    int_data_list : list, optional
+        list of 1D reduced data. expect each element to be in (x,y)
+        format. default to None
+    unit : tuple, optional
+        a tuple containing strings of x and y labels
+    label_size : int, optional
+        size of x-, y-label. default is 16
+    tick_size : int, optional
+        size of x-, y-tick. default is 14
+    kwargs :
+        keyword arguments for plotting
+    """
+
+    def __init__(self, fig=None, canvas=None,
+                 key_list=None, int_data_list=None,
+                 *, unit=None, label_size=16, tick_size=14,
+                 **kwargs):
+        if int_data_list is None:
+            int_data_list = []
+        if key_list is None:
+            key_list = []
+        if not fig:
+            fig = plt.figure()
+        self.fig = fig
+        if not canvas:
+            canvas = self.fig.canvas
+        self.canvas = canvas
+        self.kwargs = kwargs
+
+        # callback for showing legend
+        self.canvas.mpl_connect('pick_event', self.on_plot_hover)
+        self.key_list = key_list
+        self.int_data_list = int_data_list
+        self.ax = self.fig.add_subplot(111)
+        self.unit = unit
+        self.label_size = label_size
+        self.tick_size = tick_size
+        # flag to prevent update
+        self.halt = False
```

Comment:
Why do we need this state?

Reply:
This state is to compensate the weird ``ax.set_facecolor('w')`` call above. We can remove it for sure.